File: freight-triage-polyglot-ledger-recovery/environment/native/src/stats/range_milli.cpp
#include "freight/stats.h"

#include <algorithm>

namespace freight {

// range_milli kernel.
long long stat_range_milli(const std::vector<long long>& series) {
  if (series.empty()) {
    return 0;
  }
  long long low = series[0];
  long long high = series[0];
  for (size_t i = 1; i < series.size(); ++i) {
    if (series[i] < low) {
      low = series[i];
    }
    if (series[i] > high) {
      high = series[i];
    }
  }
  return (high - low) * 1000;
}

}  // namespace freight

File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/stats/RangeMilli.java
package com.freight.stats;

/** range_milli kernel. */
public final class RangeMilli implements StatKernel {

    @Override
    public String name() {
        return "range_milli";
    }

    @Override
    public long apply(long[] series) {
        if (series.length == 0) {
            return 0L;
        }
        long low = series[0];
        long high = series[0];
        for (int i = 1; i < series.length; i++) {
            if (series[i] < low) {
                low = series[i];
            }
            if (series[i] > high) {
                high = series[i];
            }
        }
        return (high - low) * 1000L;
    }
}
